ir/shared/src/main/scala/org/scalajs/ir/Types.scala

```diff
@@ -278,18 +283,18 @@ object Types {
      *  For `NullType` and `NothingType`, the names are `"null"` and
      *  `"nothing"`, respectively.
      */
-    val displayName: String = tpe match {
-      case VoidType    => "void"
-      case BooleanType => "boolean"
-      case CharType    => "char"
-      case ByteType    => "byte"
-      case ShortType   => "short"
-      case IntType     => "int"
-      case LongType    => "long"
-      case FloatType   => "float"
-      case DoubleType  => "double"
-      case NullType    => "null"
-      case NothingType => "nothing"
+    val displayName: String = (tpe.charCode: @switch) match {
```

Comment:
Instead of the match, have you considered adding `displayName` as a private field to `PrimTypeWithRef` and just forward here?

That would essentially make `PrimRef` a "view" on a PrimType, which is maybe clearer overall?